backend/src/main/java/com/google/rolecall/services/SectionServices.java

```diff
@@ -279,10 +279,40 @@ public Section editSection(SectionInfo newSection) throws EntityNotFoundExceptio
    *    in the database.
    */
   public void deleteSection(int id) throws EntityNotFoundException, InvalidParameterException {
+    checkDeleteOk(id, false);
+    Section section = getSection(id);
+
+    if(section.getType() == Section.Type.SUPER) {
+      // Check that all children are OK to delete
+      for(Position position: section.getPositions()) {
+        Integer childId = position.getSiblingId();
+        if(childId != 0 && childId > 0) {
+          Optional<Section> childQuery = sectionRepo.findById(childId);
+          if(!childQuery.isEmpty()) {
+            checkDeleteOk(childId, true);
+          }
+        }
+      }
+      // If none of the children have a deleate problem, we can delete
+      for(Position position: section.getPositions()) {
```

Comment:
I think all these extra `findById` queries can be avoided if you keep a list of childIds, append to it right after `checkDeleteOk(childId, true);`, and here just loop through that list and simply call `sectionRepo.deleteById(childId);`

Seems like that would be better performance, unless I missed something.

Reply:
I believe what I just did solves the findbyId issue.